builder/dockerfile/builder.go

```diff
@@ -240,24 +240,27 @@ func (b *Builder) build(source builder.Source, dockerfile *parser.Result) (*buil
 
 	addNodesForLabelOption(dockerfile.AST, b.options.Labels)
 
-	if err := checkDispatchDockerfile(dockerfile.AST); err != nil {
-		buildsFailed.WithValues(metricsDockerfileSyntaxError).Inc()
-		return nil, err
-	}
-
-	dispatchState, err := b.dispatchDockerfileWithCancellation(dockerfile, source)
+	stages, metaArgs, err := instructions.Parse(dockerfile.AST)
 	if err != nil {
+		if instructions.IsUnknownInstruction(err) {
+			buildsFailed.WithValues(metricsUnknownInstructionError).Inc()
+		}
```

Comment:
Don't know. I am not sure if it is a good idea to change existing metrics though. Adding a generic parsing failed metric could make sense without removing this one I suppose.